### ISSUE TITLE
FeatureEnvyRule: purpose?

### ISSUE THREAD

**Issue by maplesteve** [oclint/oclint#161]
What's exactly the purpose of the FeatureEnvyRule?

I just ran OClint with this rule on my project and it reports 172(!) violations. Nearly all of them are about messaging a Cocoa framework object. This doesn't make sense to me.

What's the point of reporting (and with that proposing to fix) the usage of `NSBundle`, `NSDictionary` and `NSUserDefaults` for the following example?

```
- (void)setUserDefaults {
 NSURL *factoryDefaultsURL = [NSBundle URLForResource:RessourceName withExtension:@"plist" subdirectory:@"" inBundleWithURL:[[NSBundle mainBundle] resourceURL]];
 NSDictionary *factoryDefaults = [NSDictionary dictionaryWithContentsOfURL:factoryDefaultsURL];
 [[NSUserDefaults standardUserDefaults] registerDefaults:factoryDefaults];
}
```

I think, `FeatureEnvyRule` has to check what type the envied class actually is and at least must not report if it's e.g. a framework class.

Though that wouldn't be enough; consider the following:

```
- (id)transformedValue:(id)item {
 return ([item class] == [Context class]) ? [item valueForKey:@"title"] : nil;
}
```

Nothing wrong with this, but a violation is reported: `Method transformedValue: messages Context more than self.`
`Context` is a class of the project and not a Cocoa framework class - `FeatureEnvyRule` has to take this into account.

Could someone please explain, what's wrong with the two code snippets and what should be done to comply with `FeatureEnvyRule`?
